src/structures/CommandInteraction.js

```diff
@@ -0,0 +1,170 @@
+    const apiMessage = content instanceof APIMessage ? content : APIMessage.create(this, content, options);
+    const { data } = await apiMessage.resolveData().resolveFiles();
+
+    await this.client.api.interactions(this.id, this.token).callback.post({
+      data: {
+        type: InteractionResponseTypes.CHANNEL_MESSAGE_WITH_SOURCE,
+        data,
+      },
+    });
+  }
+
+  /**
+   * Edits the initial reply to this interaction.
+   * @param {string|APIMessage|MessageEmbed|MessageEmbed[]} content The new content for the message
+   * @param {WebhookEditMessageOptions} [options] The options to provide
+   * @returns {Promise<?Message>}
+   */
+  async editReply(content, options) {
+    const raw = await this.webhook.editMessage('@original', content, options);
+    return this.channel?.messages.add(raw) ?? null;
+  }
+
+  /**
+   * Deletes the initial reply to this interaction.
+   * @returns {Promise<void>}
+   */
+  async deleteReply() {
+    await this.webhook.deleteMessage('@original');
+  }
+
+  /**
+   * Represents an option of a received command interaction.
+   * @property {string} name The name of the option
+   * @property {ApplicationCommandOptionType} type The type of the option
+   * @property {string|number|boolean} [value] The value of the option
+   * @property {CommandInteractionOption[]} [options] Additional options if this option is a subcommand (group)
+   * @property {User} [user] The resolved user
+   * @property {GuildMember} [member] The resolved member
+   * @property {GuildChannel} [channel] The resolved channel
+   * @property {Role} [role] The resolved role
+   * @typedef {Object} CommandInteractionOption
+   */
+
+  /**
+   * Transforms an option received from the API.
+   * @param {Object} option The received option
+   * @param {Object} resolved The resolved interaction data
+   * @returns {CommandInteractionOption}
+   * @private
+   */
+  transformOption(option, resolved) {
+    const result = {
+      name: option.name,
+      type: ApplicationCommandOptionTypes[option.type],
+    };
+
+    if (option.value) {
+      result.value = option.value;
+    }
+
+    if (option.options) {
+      result.options = option.options.map(o => this.transformOption(o, resolved));
+    }
+
+    if (option.type === ApplicationCommandOptionTypes.USER) {
+      const user = resolved.users[option.value];
+      result.user = this.client.users.add(user);
+
+      const member = resolved.members?.[option.value];
+      if (member) {
+        member.user = user;
+        result.member = this.guild.members.add(member);
+      }
+    }
+
+    if (option.type === ApplicationCommandOptionTypes.CHANNEL) {
+      const channel = resolved.channels[option.value];
+      result.channel = this.client.channels.add(channel, this.guild);
+    }
+
+    if (option.type === ApplicationCommandOptionTypes.ROLE) {
+      const role = resolved.roles[option.value];
+      result.role = this.guild.roles.add(role);
+    }
```

Comment:
Is it better to use `switch`?

Reply:
If we used `switch` here, there would need to be either an `eslint-disable` or a block for every case due to `no-case-declarations`. This would just make it unnecessarily complex.